src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -232,13 +233,20 @@ public void unexport(BodyReference ref) {
                 public Void call() throws Exception {
                     Jenkins j = Jenkins.getInstance();
                     if (paused.get() || j == null || j.isQuietingDown()) {
+                        for (FlowExecutionListener listener : FlowExecutionListener.all()) {
+                            listener.onPaused(execution);
```

Comment:
This is called whenever we are asked to schedule something, not necessarily when the pause button is pressed.